src/firefly/js/ui/DownloadOptionsDialog.jsx

```diff
@@ -28,152 +26,98 @@ export function getTypeData(key, val='', tip = '', labelV='', labelW) {
             return (
-                <div style={{marginTop: 10}}>
-                    <RadioGroupInputField
-                        options={[{label: 'Local File', value: LOCALFILE},
-                                  {label: 'Workspace', value: WORKSPACE }] }
-                        fieldKey={'fileLocation'}
+                <div style={{marginTop: 10,
+                             boxSizing: 'border-box',
+                             width: 'calc(100%)', height: 'calc(100% - 10px)',
+                             overflow: 'auto',
+                             padding: 5,
+                             border:'1px solid #a3aeb9'
+                             }}>
+                    <WorkspaceSave fieldKey={'wsSelect'}
+                                   files={wsList}
+                                   value={wsSelect}
                     />
                 </div>
             );
         };
 
-        return (
-            <div style={{height: '100%', width: '100%'}}>
-                <div>
-                    {children}
+        const showNoWSFiles = (message) => {
+            return (
+                <div style={{marginTop: 10,
+                             padding: 10,
+                             boxSizing: 'border-box',
+                             width: 'calc(100%)',
+                             textAlign: 'center',
+                             border:'1px solid #a3aeb9'}}>
+                    {message}
                 </div>
-                <ValidationField
-                    wrapperStyle={{marginTop: 10}}
-                    size={50}
-                    fieldKey={'fileName'}
-                />
+            );
+        };
 
-                {this.workspace && showLocation()}
+        return (
+            (isUpdating) ? loading() :
+                (!isEmpty(wsList) ? showSave() : showNoWSFiles('Workspace access error: ' + getWorkspaceErrorMsg()))
+        );
+    };
 
-                <div  style={{width: dialogWidth, height: dialogHeight}}>
-                    {where === WORKSPACE && showWorkspace()}
-                </div>
+    const showLocation = () => {
+        return (
+            <div style={{marginTop: 10}}>
+                <RadioGroupInputField
+                    options={[{label: 'Local File', value: LOCALFILE},
+                              {label: 'Workspace', value: WORKSPACE }] }
+                    fieldKey={'fileLocation'}
+                />
             </div>
         );
-    }
+    };
+
+    return (
+        <div style={{height: '100%', width: '100%'}}>
+            <div>
+                {children}
+            </div>
+            <ValidationField
+                wrapperStyle={{marginTop: 10}}
+                size={50}
+                fieldKey={'fileName'}
+            />
+
+            {workspace && showLocation()}
+
+            <div  style={{width: dialogWidth, height: dialogHeight}}>
+                {where === WORKSPACE && ShowWorkspace()}
+            </div>
```

Comment:
I think you misunderstood my comment on using uppercase for component name.  Take a look at this:
https://reactjs.org/docs/jsx-in-depth.html#user-defined-components-must-be-capitalized

The idea of naming functional components with uppercase is so you can use it as JSX.  i.e.
`<ShowWorkspace />`.  Otherwise, function name should be in lower camel case standard.

Reply:
Got it, that make sense. I'll change this back to showWorkspace then since it's not being used as JSX. Did you want me to change anything else in your original comment though? Or was it meant to informative? Thanks!

Reply:
Nevermind...I got it now. I can leave it with uppercase and call it as JSX instead, right?

Reply:
also just to clarify, then should I go ahead and rename other functions that are returning React components and call them using JSX as well? (loading, showSave, etc. -> Loading, ShowSave and `loading()`, `showSave()` to` <Loading />`, `<ShowSave />`)

Reply:
It has to be uppercase. Or JSX will not recognize it.